src/kafka/connection/connection_manager.cc

```diff
@@ -56,9 +58,24 @@ future<> connection_manager::disconnect(const connection_id& connection) {
 }
 
 future<metadata_response> connection_manager::ask_for_metadata(const seastar::kafka::metadata_request &request) {
-    return _connections.begin()->second->send(request);
+    return seastar::async({}, [this, request]{
+        auto it = _connections.begin();
+        metadata_response res;
+        while (it != _connections.end())
+        {
+            res =  it->second->send(request).get0();
+            if (res._error_code == error::kafka_error_code::NONE) {
+                return res;
+            }
+            it ++;
+        }
+        std::cout << _connections.size() << std::endl;
+        std::cout << "Throwing" << std::endl;
+        throw metadata_refresh_exception("No brokers responded.");
+        return res;
+    });
 }
```

Comment:
I thought we have a separate `metadata_manager` for that. And looking at other changes in this PR, this method is not even used anywhere